sympy/printing/pretty/pretty.py

```diff
@@ -1786,6 +1790,15 @@ def _print_Mul(self, product):
         else:
             args = list(product.args)
 
+        # Always show the ones out front of a product:
```

Comment:
Maybe it should be 0s then 1s then others.

Reply:
I'd rather have the args displayed in their actual order but that's not what currently happens...

Reply:
Me too. Can we carry a "dirty" label on an expression if it is created with `evaluate=False` so we know (quickly) how it was created? That way the printer could skip the ordering in that case.

Reply:
I guess we know if a Mul is unevaluated when it has more than one `Number` in the args or if the `Number` is `S.One` or is not in `args[0]` so:
```
def is_dirty(mul):
    args = mul.args
    if args[0] is S.One or any(isinstance(arg, Number) for arg in args[1:]):
        return True
```
Then in that case we can print the args multiplied precisely in order and not as a common fraction e.g.:
```julia
In [7]: Mul(1, S(1)/2, x, evaluate=False)                                                                                         
Out[7]:
  1
1⋅─⋅x
  2

In [14]: str(_)                                                                                                                   
Out[14]: '1*(1/2)*x'
```
That would also improve cases like:
```julia
In [21]: Mul(S(2)/3, S(7)/5, evaluate=False)                                                                                      
Out[21]: 
2⋅7
───
3⋅5

In [22]: str(_)                                                                                                                   
Out[22]: '2*7/(3*5)'
```

That doesn't cover all cases of unevaluated mul e.g. 2-arg mul or powers with the same exponent.